### ISSUE TITLE
Pruning with Pressure-Dependence Error

### ISSUE THREAD

**Issue by walsh-p** [ReactionMechanismGenerator/RMG-Py#490]
Here is the tail of the RMG.log file for context:

```
Pruning species [CH2]CCCCCCCCC(36)
Pruning species [CH2]CCCCCCC(=O)CCOO(2248)
Pruning species C=CCCCCCCCC(204)
Pruning species CCCCC[C](CCCC)OO(856)
Pruning species CCCCCCC(=O)C[C](C)OO(1247)
Deleting 1 empty pressure-dependent reaction networks
```

Here is the RMG.error file:

```
Traceback (most recent call last):
  File "/scratch/westgroup/RMG/RMG-Py/rmg.py", line 139, in <module>
    cProfile.runctx(command, global_vars, local_vars, stats_file)
  File "/shared/apps/python/Python-2.7.5/INSTALL/lib/python2.7/cProfile.py", line 49, in runctx
    prof = prof.runctx(statement, globals, locals)
  File "/shared/apps/python/Python-2.7.5/INSTALL/lib/python2.7/cProfile.py", line 140, in runctx
    exec cmd in globals, locals
  File "<string>", line 1, in <module>
  File "/scratch/westgroup/RMG/RMG-Py/rmgpy/rmg/main.py", line 536, in execute
    self.reactionModel.prune(self.reactionSystems, self.fluxToleranceKeepInEdge, self.maximumEdgeSpecies, self.minSpeciesExistIterationsForPrune)
  File "/scratch/westgroup/RMG/RMG-Py/rmgpy/rmg/model.py", line 1165, in prune
    nets_with_this_source = self.networkDict[source]
TypeError: unhashable type: 'list'
```

Can anyone help with this?

**Comment by KEHANG** [comment]
it seems your case is pressure-dependent case, pruning currently only safely works for non-pDep cases. We are still developing for pDep pruning.

**Comment by nickvandewiele** [comment]
@KEHANG : could you add some code that checks the input parameters and proactively deactivates pruning when it cannot be used with the given input?

**Comment by rwest** [comment]
How much harder would it be to get it working for pdep cases? (Genuine question - and the answer might be "lots", I'm not sure.)  But pruning was intended to work - from the stack trace the bug was in code that was trying to delete an empty pdep network - so such code exists, it's just broken.

**Comment by KEHANG** [comment]
Yeah, you might be right. My intension is to make it totally work for pdep in the near future. But let me first investigate this particular case and see whether it can be quickly fixed. @walsh-p could you send me input file as well as RMG.log file so that I can reproduce it?

**Comment by walsh-p** [comment]
The file has been sent to you.  Is there an efficient way to disable pruning?

**Comment by KEHANG** [comment]
I will reply you with a working input file for your situation and some instructions.:)

**Comment by alongd** [comment]
Had the same error.
Setting toleranceKeepInEdge=0 solves it for now.

**Comment by mliu49** [comment]
Fixed by #651.
